src/textual/app.py

```diff
@@ -1801,6 +1803,7 @@ def is_mounted(self, widget: Widget) -> bool:
     async def _close_all(self) -> None:
         """Close all message pumps."""
 
+        screen: Screen | Callable[[], Screen]  # Explicit type to reuse variable below.
```

Comment:
I'm surprised this doesn't blow up. `self._screen_stack` is a list of `Screens` only.

`self.SCREENS` can also contain callables, so I think that should be a new variable and typed differently.

Reply:
I don't understand why you think it would blow up! 🤷 
But I've separated the two variables.

Reply:
If `screen` really was a callable on line L1809, then I would expect that attribute to not exist. But there are no callables on the screen_stack.